geode-core/src/distributedTest/java/org/apache/geode/distributed/internal/ClusterDistributionManagerDUnitTest.java

```diff
@@ -370,6 +377,36 @@ public void testWaitForViewInstallation() {
         .untilAsserted(() -> assertThat(waitForViewInstallationDone.get()).isTrue());
   }
 
+  /**
+   * show that waitForViewInstallation works as expected when distribution manager is closed
+   * while waiting for the latest membership view to install
+   */
+  @Test
+  public void testWaitForViewInstallationDisconnectDS() {
+    InternalDistributedSystem system = getSystem();
+    ClusterDistributionManager dm = (ClusterDistributionManager) system.getDM();
+    MembershipView<InternalDistributedMember> view = dm.getDistribution().getView();
+
+    CyclicBarrier cyclicBarrier = new CyclicBarrier(2);
+    Future future = executorService.submit(() -> {
+      try {
+        cyclicBarrier.await(getTimeout().toMillis(), TimeUnit.MILLISECONDS);
+        dm.waitForViewInstallation(view.getViewId() + 1);
+      } catch (InterruptedException | BrokenBarrierException | TimeoutException e) {
+        errorCollector.addError(e);
+      }
+    });
+
+    try {
+      cyclicBarrier.await(getTimeout().toMillis(), TimeUnit.MILLISECONDS);
+      system.disconnect();
+      future.get(getTimeout().toMillis(), TimeUnit.MILLISECONDS);
+    } catch (InterruptedException | TimeoutException | ExecutionException
```

Comment:
You can delete the catch-block and use of errorCollector in the main thread. You only need to do that inside the Runnable that you submitted. From line 400 on, it's better to just let the Exceptions percolate out of the test method and let junit catch it.

Reply:
You are right! Done.